src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+    }
+
+    fn prepare_for_model(&self, input: Vec<&str>) -> Tensor {
+        let tokenized_input: Vec<TokenizedInput> = self.tokenizer.encode_list(input.to_vec(),
+                                                         128,
+                                                         &TruncationStrategy::LongestFirst,
+                                                         0);
+        let max_len = tokenized_input.iter().map(|input| input.token_ids.len()).max().unwrap();
+        let tokenized_input_tensors: Vec<tch::Tensor> = tokenized_input.
+            iter().
+            map(|input| input.token_ids.clone()).
+            map(|mut input| {
+                input.extend(vec![0; max_len - input.len()]);
+                input
+            }).
+            map(|input|
+                Tensor::of_slice(&(input))).
+            collect::<Vec<_>>();
+        Tensor::stack(tokenized_input_tensors.as_slice(), 0).to(self.var_store.device())
+    }
+
+    /// Extract entities from a text
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - `&[&str]` Array of texts to extract entities from.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<Entity>` containing extracted entities
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    ///# fn main() -> failure::Fallible<()> {
+    ///# use rust_bert::pipelines::token_classification::TokenClassificationModel;
+    ///
+    /// let ner_model =  TokenClassificationModel::new(Default::default())?;
+    /// let input = [
+    ///     "My name is Amy. I live in Paris.",
+    ///     "Paris is a city in France."
+    /// ];
+    /// let output = ner_model.predict(&input, true);
+    ///# Ok(())
+    ///# }
+    /// ```
+    pub fn predict(&self, input: &[&str], ignore_first_label: bool) -> Vec<Token> {
+        let input_tensor = self.prepare_for_model(input.to_vec());
+        let (output, _, _) = no_grad(|| {
+            self.token_sequence_classifier
+                .forward_t(Some(input_tensor.copy()),
+                           None,
+                           None,
+                           None,
+                           None,
+                           false)
+        });
+        let output = output.detach().to(Device::Cpu);
+        let score: Tensor = output.exp() / output.exp().sum1(&[-1], true, Float);
+        let labels_idx = &score.argmax(-1, true);
+
+        let mut tokens: Vec<Token> = vec!();
+        for sentence_idx in 0..labels_idx.size()[0] {
+            let labels = labels_idx.get(sentence_idx);
+            let mut word_idx: u8 = 0;
+            let mut char_offset: u16 = 0;
+            for position_idx in 0..labels.size()[0] {
+                let label_id = labels.int64_value(&[position_idx]);
+                let token = {
+                    let token_id = input_tensor.int64_value(&[sentence_idx, position_idx]);
+                    self.decode_token(token_id, label_id, &score, sentence_idx, position_idx, &mut word_idx, &input, &mut char_offset)
```

Comment:
I think the offsets should be referring to the original sentence. This would allow retrieving back the tokens with any space, special character, accents that may have been stripped and capitalization. However this process needs to happen at the tokenizer stage. As per my other comment, I would be in favor of not including this in this PR, and discuss together how to integrate it during tokenization if you are interested. What do you think?

Reply:
Yeah, I agree.. I'll roll this part back so we can keep things cleaner and implement it on the proper deeper level.

Reply:
Roll-back complete